src/olympia/users/serializers.py

```diff
@@ -12,7 +14,20 @@ class Meta:
         fields = ('id', 'name', 'url')
 
     def get_url(self, obj):
-        return absolutify(obj.get_url_path())
+        def is_developer():
+            return (self.context['is_developer']
+                    if 'is_developer' in self.context
+                    else obj.is_developer)
+
+        def is_adminish(user):
+            return user and acl.action_allowed_user(user,
+                                                    amo.permissions.USERS_EDIT)
+
+        # Only return your own profile url, and for developers.
+        current_user = getattr(
+            self.context.get('request', object()), 'user', None)
+        if obj == current_user or is_adminish(current_user) or is_developer():
+            return absolutify(obj.get_url_path())
```

Comment:
I would try to split this a bit better and write this in an easier to read style:

```python
     def get_url(self, obj):
        _is_developer() = lambda: (
            self.context['is_developer'] if 'is_developer' in self.context
            else obj.is_developer)

        _is_adminish = lambda user: (
            user if acl.action_allowed_user(user, amo.permissions.USERS_EDIT)
            else None)

        # Only return your own profile url, and for developers.
        request = self.context.get('request', None)

        user = (
            getattr(request, 'user', None) if request is not None
            else None)

        if obj == user or _is_adminish(user) or _is_developer():
            return absolutify(obj.get_url_path())
```

At least that feels easier to read to me 😇